cmd/clusterctl/api/v1alpha3/metadata.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// Metadata for a provider repository
+type Metadata struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	ReleaseSeries []ReleaseSeries `json:"releaseSeries"`
+}
+
+// ReleaseSeries mapa a provider release series (major/minor) with a ClusterAPIVersion
+type ReleaseSeries struct {
+	// +optional
+	Major uint `json:"major,omitempty"`
+
+	// +optional
+	Minor uint `json:"minor,omitempty"`
+
+	// ClusterAPIVersion indicates the cluster API supported version.
+	// +optional
+	ClusterAPIVersion string `json:"clusterAPIVersion,omitempty"`
```

Comment:
Should we make this a slice so that we can potentially allow for compatibility with multiple cluster-api versions?

Also, is there a reason for this to be optional vs required?

Reply:
removed optional

> Should we make this a slice so that we can potentially allow for compatibility with multiple cluster-api versions?

@detie being this the first try to manage updates both in providers and in clusterctl I'm leaning toward keep it simple. Nevertheless but I'm going to open an issue to discuss this point; Afterward I will eventually rework accordingly to the outcomes.

Reply:
+1 on targeting only a single version for now